docs/develop/go/message-passing.mdx

```diff
@@ -192,309 +318,266 @@ if err != nil {
-			counter += arg.Add
-			result := YourUpdateResult{
-				Total: counter,
-			}
-			return result, nil
-		},
-		// Set the isPositive validator.
-		workflow.UpdateHandlerOptions{Validator: isPositive},
-	)
-	if err != nil {
-		return WFResult{}, err
-	}
-	if err := workflow.Sleep(ctx, time.Minute); err != nil {
-		return WFResult{}, err
-	}
-	return WFResult{Total: counter}, nil
-}
+    - A failed Child Workflow
+    - A failed Activity (if the activity retries have been set to a finite number)
+    - The Workflow author returning an `error`
+    - A panic in the handler, depending on the `WorkflowPanicPolicy`.
 
-// isPositive is a validator function.
-// It returns an error if the int value is below 1.
-// This function can not change the state of the Workflow.
-// workflow.Context can be used to log
-func isPositive(ctx workflow.Context, u YourUpdateArg) error {
-	log := workflow.GetLogger(ctx)
-	if u.Add < 1 {
-		log.Debug("Rejecting non-positive number, positive integers only", "UpdateValue", u.Add)
-		return fmt.Errorf("addend must be a positive integer (%v)", u.Add)
-	}
-	log.Debug("Accepting Update", "UpdateValue", u.Add)
-	return nil
-}
-```
+- **The handler caused the Workflow Task to fail.**
+  A [Workflow Task Failure](/references/failures) causes the server to retry Workflow Tasks indefinitely. What happens to your Update request depends on its stage:
+    - If the request hasn't been accepted by the server, you receive a [`FAILED_PRECONDITION`](https://pkg.go.dev/go.temporal.io/api/serviceerror#FailedPrecondition) error.
+    - If the request has been accepted, it is durable.
+      Once the Workflow is healthy again after a code deploy, use a [`WorkflowUpdateHandle`](https://pkg.go.dev/go.temporal.io/sdk/client#WorkflowUpdateHandle) to fetch the Update result.
 
-### Send an Update from a Temporal Client {#send-update-from-client}
+- **The Workflow finished while the Update handler execution was in progress**:
+  You'll receive a [`ServiceError`](https://pkg.go.dev/go.temporal.io/api/serviceerror#ServiceError) "workflow execution already completed"`.
 
-**How to send an Update from a Temporal Client using the Go SDK.**
+   This will happen if the Workflow finished while the Update handler execution was in progress, for example because
 
-Invoke the `UpdateWorkflow()` method on an instance of the [Go SDK Temporal Client](https://pkg.go.dev/go.temporal.io/sdk/client#Client) to dispatch an [Update](/encyclopedia/workflow-message-passing#sending-updates) to a Workflow Execution.
+    - The Workflow was canceled or failed.
 
-You must provide the Workflow Id, but specifying a Run Id is optional.
-If you supply only the Workflow Id (and provide an empty string as the Run Id param), the currently running Workflow Execution receives the Update.
+    - The Workflow completed normally or continued-as-new and the Workflow author did not [wait for handlers to be finished](/encyclopedia/workflow-message-passing#finishing-message-handlers).
```

Comment:
I don't think the word "author" is correct here

Reply:
Hm, can you explain why?